src/nbl/asset/utils/CIESProfileParser.cpp

```diff
@@ -0,0 +1,132 @@
+    }
+    else if (line != "TILT=NONE") {
+        errorMsg = "TILT not specified";
+        return false;
+    }
+
+    int numLamps = getInt("numLamps truncated");
+    double lumensPerLamp = getDouble("lumensPerLamp truncated");
+    double candelaMultiplier = getDouble("candelaMultiplier truncated");
+    int vSize = getInt("vSize truncated");
+    int hSize = getInt("hSize truncated");
+
+    int type_ = getInt("type truncated");
+    if (error)
+        return false;
+    if (type_ <= 0 || type_ > 3) {
+        errorMsg = "unrecognized type";
+        return false;
+    }
+    CIESProfile::PhotometricType type =
+        static_cast<CIESProfile::PhotometricType>(type_);
+    if (type != CIESProfile::PhotometricType::TYPE_C) {
+        errorMsg = "Only type C is supported for now";
+        return false;
+    }
+
+    int unitsType = getInt("unitsType truncated");
+    double width = getDouble("width truncated"),
+        length = getDouble("length truncated"),
+        height = getDouble("height truncated");
+    double ballastFactor = getDouble("ballastFactor truncated");
+    double reserved = getDouble("reserved truncated");
+    double inputWatts = getDouble("inputWatts truncated");
+    if (error)
+        return false;
+
+    result = CIESProfile(type, hSize, vSize);
+    auto& vAngles = result.vAngles;
+    for (int i = 0; i < vSize; i++) {
+        vAngles[i] = getDouble("vertical angle truncated");
+    }
+    if (!std::is_sorted(vAngles.begin(), vAngles.end())) {
+        errorMsg = "Angles should be sorted";
+        return false;
+    }
+    if (vAngles[0] != 0.0 && vAngles[0] != 90.0) {
+        errorMsg = "First angle must be 0 or 90 in type C";
+        return false;
+    }
+    if (vAngles[vSize - 1] != 90.0 && vAngles[vSize - 1] != 180.0) {
+        errorMsg = "Last angle must be 90 or 180 in type C";
+        return false;
+    }
+
+    auto& hAngles = result.hAngles;
+    for (int i = 0; i < hSize; i++) {
+        hAngles[i] = getDouble("horizontal angle truncated");
+        if (i != 0 && hAngles[i - 1] > hAngles[i])
+            return false; // Angles should be sorted
+    }
+
+    double factor = ballastFactor * candelaMultiplier;
+    for (int i = 0; i < hSize; i++) {
+        for (int j = 0; j < vSize; j++) {
+            result.setValue(i, j, factor * getDouble("intensity value truncated"));
+        }
+    }
+    /*
+    if (hAngles.back() == 180.0) {
+        for (int i = (int)hAngles.size() - 2; i >= 0; i--) {
+            result.addHoriAngle(360.0 - hAngles[i]);
+            for (int j = 0; j < vSize; j++)
+                result.setValue(result.hAngles.size() - 1, j, result.getValue(i, j));
+        }
+    }
+    */
+
+    // to interprete angles pairs starting with (0, phi) for proper computations we shift the angles
+    result.hAOffset = result.hAngles[0];
+    if (result.hAOffset != 0)
+        for (auto& hAngle : result.hAngles)
+            hAngle = std::fmod((hAngle - result.hAOffset + 360), 360);
```

Comment:
this makes no sense?

why are you adding 360? why are you going module 360?

aren't angles outside of 0,359.999 banned?

Reply:
> For Type C photometry, the first value shall always
be 0 degrees, and the last value shall be one of the
following:
1) 0 - in this case, there is only one horizontal angle,
and the luminaire is assumed to be laterally
symmetric in all planes.
2) 90 - the luminaire is assumed to be symmetric in
each quadrant.
3) 180 - the luminaire is assumed to be symmetric
about the 0 to 180 degree plane.
4) 360 - in this case, the luminaire is assumed to
exhibit no lateral symmet

Reply:
do validation of this

Only if the version is 1995, allow also start at 90 and end at 270, assume symmetry about the 90 to 270 degree plane

Reply:
need loader to spit out ICPUImageView + Sampler Wrap modes as metadata

Reply:
resolved in following commit